### PR TITLE
Reduce code duplication between metadata and lite connectors, allow non-enum connector properties

### PR DESCRIPTION
**Pull request by misolt**
This is part of an effort to extract common snowflake connector code into abstract and util classes. Some of the changes were left out for later to keep this PR at a reviewable size.

Main changes:
- Allow connectors to have non-enum connector properties
- Combine the assessment-specific tasks of metadata and lite connectors
- Remove custom parsing of enum names
- Fix bug where table-storage metrics were referred to only as 'table' in descriptions, caused by the parsing mentioned above
- Move parts of the metadata connector to the abstract connector
- Extract parts of the metadata connector to separate classes

### CODE REVIEW

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakeTaskUtil.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.snowflake;
+
+import com.google.edwmigration.dumper.application.dumper.task.AbstractJdbcTask;
+import com.google.edwmigration.dumper.application.dumper.task.JdbcSelectTask;
+import com.google.edwmigration.dumper.application.dumper.task.Summary;
+
+final class SnowflakeTaskUtil {
+
+  static AbstractJdbcTask<Summary> fromVariant(
```
Comment (dominik-klimczak) [inline dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakeTaskUtil.java:25]: Why do we create variants and convert them to tasks separately? It seems it's just extra complexity. Can't we have a TaskFactory that will create JdbcSelectTask with or without filter?

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakePlanner.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import com.google.common.collect.ImmutableList;
+import com.google.edwmigration.dumper.application.dumper.connector.ResultSetTransformer;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.SnowflakeMetadataDumpFormat.ExternalTablesFormat;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.SnowflakeMetadataDumpFormat.FunctionInfoFormat;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.SnowflakeMetadataDumpFormat.TableStorageMetricsFormat;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.SnowflakeMetadataDumpFormat.WarehousesFormat;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+@ParametersAreNonnullByDefault
+final class SnowflakePlanner {
```
[the remark this answers is not in the document]
Comment (misolt) [reply dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakePlanner.java:38]: Done, I added a short summary.

The idea is to have a helper class for generating lists of tasks to reduce complexity of the connector service.

It was similar for Oracle, the difference is that here the helper generates only fragments of the tasks list, which can be shared between different connectors.

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakeTaskUtil.java (hunk 1 above)
Comment (misolt) [reply dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakeTaskUtil.java:25]: Acknowledged, will fix in future PR

The original authors wanted to have a class that represents a task but without the main part of the query - this main part is provided by the `format` argument. For now I only moved this class from the connector to a separate file and made it AutoValue, but in the long term I also would like to remove TaskVariant.

Comment (misolt) [reply dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakeTaskUtil.java]: I removed `TaskVariant`

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakePlanner.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+import com.google.common.collect.ImmutableList;
+import com.google.edwmigration.dumper.application.dumper.connector.ResultSetTransformer;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.SnowflakeMetadataDumpFormat.ExternalTablesFormat;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.SnowflakeMetadataDumpFormat.FunctionInfoFormat;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.SnowflakeMetadataDumpFormat.TableStorageMetricsFormat;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.SnowflakeMetadataDumpFormat.WarehousesFormat;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+/**
+ * Generates task lists for Snowflake connectors.
```
Comment (dominik-klimczak) [inline dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakePlanner.java:32]: Generator of task lists for Snowflake connectors.

(noun phrase is preferred for class javadoc)

**Review by dominik-klimczak — APPROVED** [review]

File: dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakePlanner.java (hunk 3 above)
Comment (misolt) [reply dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakePlanner.java]: Done

**Pull request merged**